### DISCUSSION TITLE
MapboxOverlay interleaved and onClick argument

### DISCUSSION THREAD

**Discussion by kaorukitao** [visgl/deck.gl#8885] [Q&A] [answered]
I have some questions on MapboxOverlay having layers of PointCloudLayer instance. I use MapLibre-GL 4.2.0 and I am trying display point cloud with deck.gl 9.0.14.
I wrote like this;

```JavaScript
const deckOverlay = new MapboxOverlay({
  interleaved: true,
  layers: [
    new PointCloudLayer({
      id: "test",
      data: {
        length: n,
        attributes: { ... },
      },
      material: false,
      pointSize: 10,
      coordinateOrigin: [ latitude, longitude ],
      coordinateSystem: COORDINATE_SYSTEM.METER_OFFSETS,
      getNormal: [ 0, 0, 1 ],
      pickable: true,
    })
  ],
  onClick: ( d ) => {
    console.log( d );
  }
});
```

### First question:

Changing interleaved value to false, points were not displayed. Why ?
When I create a similar program MapLibre-GL 3.6.2 and deck.gl 8.9.32, points were displayed whether interleaved is both true and false.

### Second question:

Tell me aboud "onClick" argument "d". "d" has a property "color". I thought it is the color of clicked point but it seems different. When I draw 3 points, first point color is Uint8Array [ 1, 0, 0, 2 ], second is [ 2, 0, 0, 2 ], third is [ 3, 0, 0, 2 ].
What value does property "color" indicate ?

### Third question:

"d" mentioned above has a property "index". Does it show the order of clicked point ?

**Comment by Pessimistress** [comment]
> coordinateOrigin: [ latitude, longitude ]

All deck.gl cordinates are in [longitude, latitude] order.

> Changing interleaved value to false, points were not displayed. Why ?

Make sure you have Maplibre's CSS stylesheet attached.

> Tell me aboud "onClick" argument

[PickingInfo](https://deck.gl/docs/developer-guide/interactivity#the-pickinginfo-object)

**Reply by kaorukitao** [reply]
Thank you.

I typed `coordinateOrigin` wrong. In my script, it was correct.

>>Changing interleaved value to false, points were not displayed. Why ?

>Make sure you have Maplibre's CSS stylesheet attached.

Points were not displayed whether Maplibre's CSS stylesheet attached or not.
I checked canvases with developer tool of safari. I found map canvas and blank canvas.  

PickingInfo documents does not tell me what color is.
Maybe `color[0]` and `color[1]` and `color[2]` represent object number and `color[3]` represents layer number ?

**Reply by Pessimistress** [reply]
`color` is not meant for user consumption. If you want to know how it works, you can read https://deck.gl/docs/developer-guide/custom-layers/picking#how-it-works

> Points were not displayed whether Maplibre's CSS stylesheet attached or not.

Hard to say what is wrong without seeing your full code. You can check out https://github.com/visgl/deck.gl/tree/master/examples/get-started/pure-js/maplibre for a working example.

**Reply by kaorukitao** [reply, marked as answer]
I found the cause of problem.

**not displayed**
```
const map = new MapLibreGL.Map({ ... });
map.once( "load", () => {
  const deckOverlay= new MapboxOverlay({
    interleaved: false,
    layers: []
  });
  map.addControl( deckOverlay );
  //create layers using deckOverlay._deck.device.createBuffer()
});
```
Maybe the timing to create `deckOverlay._deck.device`.

**displayed**
```
const map = new MapLibreGL.Map({ ... });
const deckOverlay= new MapboxOverlay({
  interleaved: false,
  layers: []
});
map.addControl( deckOverlay );
map.once( "load", () => {
  //create layers using deckOverlay._deck.device.createBuffer()
});
```

I tried to investigate the difference between pointed sample and my script, and found.
Thank you very much.
